src/pudl/transform/eia861.py

```diff
@@ -193,79 +681,773 @@ def service_territory(raw_dfs, tfr_dfs):
+            name. It will be mutated by this function.
+
+    Returns:
+        dict: A dictionary of transformed EIA 861 dataframes, keyed by table name.
+
+    """
+    idx_cols = [
+        'utility_id_eia',
+        'state',
+        'balancing_authority_code_eia',
+        'report_date',
+    ]
+
+    # Pre-tidy clean specific to sales table
+    raw_nm = tfr_dfs["net_metering_eia861"].copy()
+
+    ###########################################################################
+    # Tidy Data:
+    ###########################################################################
+
+    logger.info("Tidying the EIA 861 Net Metering table.")
+    # Normalize by customer class (must be done before normalizing by fuel class)
+    tidy_nm, idx_cols = _tidy_class_dfs(raw_nm, 'Net Metering',
+                                        idx_cols, CUSTOMER_CLASSES, 'customer_class')
+
+    # Normalize by fuel class
+    tidy_nm, idx_cols = _tidy_class_dfs(tidy_nm, 'Net Metering',
+                                        idx_cols, TECH_CLASSES, 'tech_class', keep_totals=True)
+
+    # No duplicates to speak of but take measures to check just in case
+    _check_for_dupes(tidy_nm, 'Net Metering', idx_cols)
+
+    # No transformation needed
+
+    tfr_dfs["net_metering_eia861"] = tidy_nm
+
+    return tfr_dfs
+
+
+def non_net_metering(tfr_dfs):
+    """
+    Transform the EIA 861 Non-Net Metering table.
+
+    Args:
+        tfr_dfs (dict): A dictionary of transformed EIA 861 DataFrames, keyed by table
+            name. It will be mutated by this function.
+
+    Returns:
+        dict: A dictionary of transformed EIA 861 dataframes, keyed by table name.
+
+    """
+    idx_cols = [
+        'utility_id_eia',
+        'state',
+        'balancing_authority_code_eia',
+        'report_date',
+    ]
+
+    # Pre-tidy clean specific to sales table
+    raw_nnm = tfr_dfs["non_net_metering_eia861"].copy()
+
+    ###########################################################################
+    # Tidy Data:
+    ###########################################################################
+
+    # Normalize by customer class (must be done before normalizing by fuel class)
+    tidy_nnm, idx_cols = _tidy_class_dfs(raw_nnm, 'Non Net Metering',
+                                         idx_cols, CUSTOMER_CLASSES, 'customer_class', keep_totals=True)
+
+    # Normalize by fuel class
+    tidy_nnm, idx_cols = _tidy_class_dfs(tidy_nnm, 'Non Net Metering',
+                                         idx_cols, TECH_CLASSES, 'tech_class', keep_totals=True)
+
+    # No duplicates to speak of but take measures to check just in case
+    _check_for_dupes(tidy_nnm, 'Non Net Metering', idx_cols)
+
+    # Delete total_capacity_mw col for redundancy (it doesn't matter which one)
+    tidy_nnm = (
+        tidy_nnm.drop(columns='capacity_mw_y')
+        .rename(columns={'capacity_mw_x': 'capacity_mw'})
+    )
```

Comment:
I assuming these double capacity columns (with the `_x`/`_y` suffixes) come in during the second use of `_tidy_class_dfs`. Is that right? Is there any other overlapping columns? I assume you already checked for this, but just wanted to flag it.

Reply:
No other overlapping columns. This is the column that I previously deleted from the column map and it caused me problems.